lib/workload/stateful/securitygroup/component.ts

```diff
@@ -1,25 +1,33 @@
 import { Construct } from 'constructs';
-import { IVpc, SecurityGroup } from 'aws-cdk-lib/aws-ec2';
-import { aws_ec2 } from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
 
 export interface SecurityGroupProps {
   securityGroupName: string;
   securityGroupDescription: string;
 }
-
+export interface SecurityGroupConstructProps extends SecurityGroupProps {
+  dbSecurityGroup: ec2.SecurityGroup;
+}
 export class SecurityGroupConstruct extends Construct {
-  constructor(scope: Construct, id: string, vpc: IVpc, props: SecurityGroupProps) {
+  constructor(scope: Construct, id: string, vpc: ec2.IVpc, props: SecurityGroupConstructProps) {
     super(scope, id);
 
```

Comment:
Just wondering here, if the SG is purposely for lambda, would it be better if this is created right on the stack where lambda will be created (on the `OrcaBusStatelessStack` before the workflow construct)?

Reply:
You are right, there will be another App specific SG down there in Stateless Construct for other inbound/outbound rules need. 

However. This SG is special for Lambda to Database with VPC. Because the DB will be placed in Isolated subnet i.e. air-gapped. No direct access from Internet nor external but must be within VPC only.